java/client/src/main/java/glide/internal/GlideCoreClient.java

```diff
@@ -0,0 +1,1100 @@
+
+        // Use the existing native handle
+        this.nativeClientHandle.set(existingHandle);
+
+        // Create shared state for proper cleanup coordination
+        this.nativeState = new NativeState(existingHandle);
+
+        // Register cleanup action with Cleaner - but don't double-close since handle is managed
+        // externally
+        this.cleanable = CLEANER.register(this, new CleanupAction(this.nativeState));
+    }
+
+    /** Convert BaseClientConfiguration to our internal Config format */
+    private static Config convertFromBaseClientConfiguration(
+            glide.api.models.configuration.BaseClientConfiguration config) {
+        // Extract addresses from NodeAddress objects
+        List<String> addresses =
+                config.getAddresses().stream()
+                        .map(addr -> addr.getHost() + ":" + addr.getPort())
+                        .collect(java.util.stream.Collectors.toList());
+
+        Config result = new Config(addresses);
+
+        // Map configuration fields
+        result.useTls(config.isUseTLS());
+        result.useInsecureTls(resolveInsecureTls(config));
+
+        if (config.getCredentials() != null) {
+            glide.api.models.configuration.ServerCredentials creds = config.getCredentials();
+            result.credentials(creds.getUsername(), creds.getPassword());
+        }
+
+        if (config.getRequestTimeout() != null) {
+            result.requestTimeout(config.getRequestTimeout());
+        }
+
+        result.connectionTimeout(resolveConnectionTimeout(config));
+
+        if (config.getProtocol() != null) {
+            result.protocol(config.getProtocol());
+        }
+
+        if (config.getClientName() != null) {
+            result.clientName(config.getClientName());
+        }
+
+        if (config.getReconnectStrategy() != null) {
+            result.reconnectStrategy(config.getReconnectStrategy());
+        }
+
+        // Handle cluster vs standalone specific configurations
+        if (config instanceof glide.api.models.configuration.GlideClusterClientConfiguration) {
+            glide.api.models.configuration.GlideClusterClientConfiguration clusterConfig =
+                    (glide.api.models.configuration.GlideClusterClientConfiguration) config;
+            result.clusterMode(true);
+
+            if (clusterConfig.getReadFrom() != null) {
+                result.readFrom(clusterConfig.getReadFrom());
+            }
+        } else if (config instanceof glide.api.models.configuration.GlideClientConfiguration) {
+            glide.api.models.configuration.GlideClientConfiguration standaloneConfig =
+                    (glide.api.models.configuration.GlideClientConfiguration) config;
+            result.clusterMode(false);
+
+            if (standaloneConfig.getDatabaseId() != null) {
+                result.databaseId(standaloneConfig.getDatabaseId());
```

Comment:
DatabaseId is now part of BaseClientConfiguration and should apply to both cluster and standalone modes. Please rebase or merge if that's not the case.